### DISCUSSION TITLE
[NODRIVER] I cannot for the life of me get nodriver to run!

### DISCUSSION THREAD

**Discussion by tomdav999** [ultrafunkamsterdam/undetected-chromedriver#2090] [Q&A] [answered]
I cannot for the life of me get nodriver to run (Windows 10 with fresh install of python and nodriver).  Very simple script.  It throws an error as soon as it hits  `browser = await uc.start() `

What am I doing wrong?

```
import nodriver as uc

async def main():
	print('before')
	browser = await uc.start()
	print('after')

if __name__ == "__main__":
    uc.loop().run_until_complete(main())
```

Output: 

> py test.py
> before
> Traceback (most recent call last):
>   File "C:\Users\XXX\documents\YYY\test.py", line 11, in <module>
>     uc.loop().run_until_complete(main())
>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
>   File "C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\asyncio\base_events.py", line 721, in run_until_complete
>     return future.result()
>            ~~~~~~~~~~~~~^^
>   File "C:\Users\XXX\documents\YYY\test.py", line 5, in main
>     browser = await uc.start()
>               ^^^^^^^^^^^^^^^^
>   File "C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\site-packages\nodriver\core\util.py", line 95, in start
>     return await Browser.create(config)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\site-packages\nodriver\core\browser.py", line 90, in create
>     await instance.start()
>   File "C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\site-packages\nodriver\core\browser.py", line 393, in start
>     await self.connection.send(cdp.target.set_discover_targets(discover=True))
>   File "C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\site-packages\nodriver\core\connection.py", line 405, in send
>     if not self.websocket or self.closed:
>                              ^^^^^^^^^^^
>   File "C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\site-packages\nodriver\core\connection.py", line 361, in __getattr__
>     return getattr(self.target, item)
> AttributeError: 'NoneType' object has no attribute 'closed'
> successfully removed temp profile C:\Users\XXX\AppData\Local\Temp\uc_ozulbtqk
> Task was destroyed but it is pending!
> task: <Task pending name='Task-5' coro=<Connection.aclose() running at C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\site-packages\nodriver\core\connection.py:298>>
> C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\asyncio\base_events.py:746: RuntimeWarning: coroutine 'Connection.aclose' was never awaited
> Exception ignored in: <function BaseSubprocessTransport.__del__ at 0x0000018C2B848360>
> Traceback (most recent call last):
>   File "C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\asyncio\base_subprocess.py", line 130, in __del__
>   File "C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\asyncio\base_subprocess.py", line 107, in close
>   File "C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\asyncio\proactor_events.py", line 109, in close
>   File "C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\asyncio\base_events.py", line 829, in call_soon
>   File "C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\asyncio\base_events.py", line 552, in _check_closed
> RuntimeError: Event loop is closed
> Task was destroyed but it is pending!
> task: <Task pending name='Task-3' coro=<Connection.keepalive() running at C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\site-packages\websockets\asyncio\connection.py:797> wait_for=<Future pending cb=[Task.task_wakeup()]>>
> Task was destroyed but it is pending!
> task: <Task pending name='Task-4' coro=<Listener.listener_loop() running at C:\Users\XXX\AppData\Local\Programs\Python\Python313\Lib\site-packages\nodriver\core\connection.py:595> wait_for=<Future pending cb=[Task.task_wakeup()]>>

**Comment by tomdav999** [comment, marked as answer]
If anyone else has the same problem and doesn't want to waste half a day trying to figure it out, apparently you need to downgrade websockets:

`pip install websockets==13.1`

https://github.com/ultrafunkamsterdam/undetected-chromedriver/issues/2069

**Reply by fg0611** [reply]
I had not that issue but I'm trying to scrape linkedin the freaking modal doesnt go away. Its a nodriver thing, cand click on the close btn nor click outside the modal

**Reply by ultrafunkamsterdam** [reply]
> I had not that issue but I'm trying to scrape linkedin the freaking modal doesnt go away. Its a nodriver thing, cand click on the close btn nor click outside the modal

Has nothing to do with nodriver. Is your issue . If you can't get a modal away and don't want to investigate, just remove it's html and backdrop with js for example

**Reply by fg0611** [reply]
I found the way to do it but not reading the library docs I had to read articles about how the library works to solve my issue.
Maybe by adding more code examples on the basics could help.

Thanks for your work and effort. I wish I had your capacity to create stuff like nodrive.
